### PR TITLE
HEELDS-1026 Check if centre emails are duplicate before submitting My…

### PR DESCRIPTION
**Pull request by simonbrent**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-1026

### Description
Fixed saving My Account > Edit Details form when the user has not chosen a centre and tries to change centre-specific email addresses to ones which are already in use at that centre

### Screenshots
edit_details

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Comment by SteveJacksonSoft** [comment]
Oh, I wanted to request a change to the validation message while we're here. Users may interpret "This email is already in use" as "you're using this centre email at another centre". Can we clarify it and say, "This email is in already use by another user at the centre"?

File: DigitalLearningSolutions.Web/Controllers/MyAccountController.cs [hunk 1]
```diff
@@ -208,34 +208,35 @@ DlsSubApplication dlsSubApplication
...
 
-            if (
-                centreId.HasValue && !string.IsNullOrWhiteSpace(formData.CentreSpecificEmail) &&
-                userDataService.CentreSpecificEmailIsInUseAtCentreByOtherUser(
-                    formData.CentreSpecificEmail,
-                    centreId.Value,
-                    userId
-                )
-            )
+            var centreSpecificEmails = centreId.HasValue
```
[the remark this answers is not in the document]
Comment (simonbrent) [reply DigitalLearningSolutions.Web/Controllers/MyAccountController.cs]: Done

**Comment by simonbrent** [comment]
I have also changed the error message when centre-specific email is already in use in both the MyAccountController, and the in EditDelegateController

**Review by SteveJacksonSoft — APPROVED** [review]
Beautiful

**Pull request merged**